File: openapi-test-output-java-jaxrs/testapi/api/PetsApi.java
package testapi.api;

import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.DELETE;
import jakarta.ws.rs.DefaultValue;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.QueryParam;
import jakarta.ws.rs.core.MediaType;
import java.lang.Void;
import java.util.List;
import java.util.Optional;
import testapi.model.Pet;
import testapi.model.PetCreate;

@Path("/pets")
public sealed interface PetsApi {
  @POST
  @Path("/")
  @Consumes(MediaType.APPLICATION_JSON)
  @Produces(MediaType.APPLICATION_JSON)
  /** Create a pet */
  CreatePetResponse createPet(PetCreate body);

  @DELETE
  @Path("/{petId}")
  /** Delete a pet */
  Void deletePet(
  
    /** The pet ID */
    @PathParam("petId") String petId
  );

  @GET
  @Path("/{petId}")
  @Produces(MediaType.APPLICATION_JSON)
  /** Get a pet by ID */
  GetPetResponse getPet(
  
    /** The pet ID */
    @PathParam("petId") String petId
  );

  @GET
  @Path("/")
  @Produces(MediaType.APPLICATION_JSON)
  /** List all pets */
  List<Pet> listPets(
    /** Maximum number of pets to return */
    @QueryParam("limit") @DefaultValue("20") Optional<Integer> limit,
    /** Filter by status */
    @QueryParam("status") @DefaultValue("available") Optional<String> status
  );
}

File: openapi-test-kotlin-server-quarkus-reactive-client-mp-reactive/testapi/api/PetsApi.kt
package testapi.api

import com.fasterxml.jackson.databind.JsonNode
import io.smallrye.mutiny.Uni
import java.lang.Void
import java.util.Optional
import kotlin.collections.List
import testapi.model.Pet
import testapi.model.PetCreate

sealed interface PetsApi {
  /** Create a pet */
  fun createPet(body: PetCreate): Uni<CreatePetResponse>

  /** Delete a pet */
  fun deletePet(
    /** The pet ID */
    petId: String
  ): Uni<DeletePetResponse>

  /** Get a pet by ID */
  fun getPet(
    /** The pet ID */
    petId: String
  ): Uni<GetPetResponse>

  /** Get pet photo */
  fun getPetPhoto(
    /** The pet ID */
    petId: String
  ): Uni<Void>

  /** List all pets */
  fun listPets(
    /** Maximum number of pets to return */
    limit: Optional<Integer>,
    /** Filter by status */
    status: Optional<String>
  ): Uni<List<Pet>>

  /** Upload a pet photo */
  fun uploadPetPhoto(
    /** The pet ID */
    petId: String,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array<Byte>
  ): Uni<JsonNode>
}